src/GraphQl/Resolver/Stage/ReadStage.php

```diff
@@ -157,7 +157,11 @@ private function getNormalizedFilters(array $args): array
 
             if (\is_string($name) && strpos($name, $this->nestingSeparator)) {
                 // Gives a chance to relations/nested fields.
-                $filters[str_replace($this->nestingSeparator, '.', $name)] = $value;
+                $index = array_search($name, array_keys($filters), true);
+                $filters =
+                    \array_slice($filters, 0, $index + 1) +
+                    [str_replace($this->nestingSeparator, '.', $name) => $value] +
+                    \array_slice($filters, $index);
```

Comment:
Why starting at index?

Reply:
It does not make a difference in the output as the duplicated key gets merged so I did not notice that this code adds it twice. Will add `+1` there as well.